src/main/java/frc/robot/subsystems/Elevator.java

```diff
@@ -0,0 +1,244 @@
+    }
+
+    /**
+     * Stops the elevator and sets the goal to the current setpoint
+     */
+    private void limitReached() {
+        cancelAllElevatorCommands();
+        m_controller.forceAtGoal();
+        forcedGoal = true;
+    }
+
+    /**
+     * Calculates and sets the profiled speed of the motor
+     */
+    private void calculateSpeed() {
+        totalSpeed = m_controller.calculate(m_leftEncoder.getPosition())
+                     + m_feedforward.calculate(m_controller.getSetpoint().velocity);
+        setLeftMotorSpeed(totalSpeed);
+    }
+
+    /**
+     * Checks for limits and sets the motor speed
+     */
+    private void setProfiledSpeed() {
+        if (!forcedGoal && 
+           (checkLimitSwitchPress(m_forwardLimit) || 
+            checkLimitSwitchPress(m_reverseLimit))) {
+
+            System.out.println(checkLimitSwitchPress(m_forwardLimit) 
+                ? "Top Limit Hit" : "Bottom Limit Hit");
+
+            limitReached();
+        } else {
+            calculateSpeed();
+            if (!this.checkLimitSwitchPress(m_forwardLimit) 
+             && !this.checkLimitSwitchPress(m_reverseLimit)) {
+                forcedGoal = false;
+            } 
+        }
+    }
+
+    /**
+     * Cancels all elevator commands
+     */
+    private void cancelAllElevatorCommands() {
+        CommandScheduler.getInstance().cancel(this.moveToPositionCommand(kElevatorPositions.SOURCE));
+        CommandScheduler.getInstance().cancel(this.moveToPositionCommand(kElevatorPositions.AMP));
+        CommandScheduler.getInstance().cancel(this.moveToPositionCommand(kElevatorPositions.INTAKE));
+    }
+
+    /**
+     * Sets the elevator positions
+     * @param position the position to be set
+     * @return the command to get to the position
+     */
+    public Command moveToPositionCommand(ElevatorConstants.kElevatorPositions position) {
+        switch (position) {
+            case AMP:
+                return this.runOnce(() -> m_controller.setGoal(
+                    ElevatorConstants.kElevatorGoals[
+                    ElevatorConstants.kElevatorPositions.AMP.ordinal()]));
+
+            case SOURCE:
+                return this.runOnce(() -> m_controller.setGoal(
+                    ElevatorConstants.kElevatorGoals[
+                    ElevatorConstants.kElevatorPositions.SOURCE.ordinal()]));
+
+            case INTAKE:
+            System.out.println("in elev");
+                return this.runOnce(() -> m_controller.setGoal(
+                    ElevatorConstants.kElevatorGoals[
+                    ElevatorConstants.kElevatorPositions.INTAKE.ordinal()]));
+
+            default:
+                return null;
+        }
+    }
+
+    /**
+     * Stops the elevator
+     * @return the command for stopping the elevator
+     */
+    public Command stopElevatorCommand() {
+        return this.runOnce(() -> cancelAllElevatorCommands());
+    }
+    
+    private void putToDashboard() {
```

Comment:
Lots of commented out lines of code here, if not needed in the future please remove. If needed move specific dashboard calls to a new method that can be invoked when needed.